woocommerce/api/Abstract_Cacheable_API_Base.php

```diff
@@ -0,0 +1,228 @@
+		 * a non-cacheable request is accidentally flagged as cacheable.
+		 *
+		 * @since 5.10.10
+		 * @param bool $is_cacheable whether the request is cacheable
+		 * @param SV_WC_API_Request $request the request instance
+		 */
+		return (bool) apply_filters( 'wc_plugin_' . $this->get_plugin()->get_id() . '_api_request_is_cacheable', true, $this->get_request() );
+	}
+
+
+	/**
+	 * Gets the cache lifetime for the current request.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return int
+	 */
+	protected function get_request_cache_lifetime() : int {
+
+		/**
+		 * Filters API request cache lifetime.
+		 *
+		 * Allows actors to override cache lifetime for cacheable API requests. This may be useful for debugging
+		 * API requests by temporarily setting short cache timeouts.
+		 *
+		 * @since 5.10.10
+		 * @param int $lifetime cache lifetime in seconds, 0 = unlimited
+		 * @param SV_WC_API_Request $request the request instance
+		 */
+		return (int) apply_filters( 'wc_plugin_' . $this->get_plugin()->get_id() . '_api_request_cache_lifetime' , $this->get_request()->get_cache_lifetime(), $this->get_request() );
+	}
+
+
+
+	/**
+	 * Determine whether the response was loaded from cache or not.
+	 *
+	 * @since 5.10.10
+	 * @return bool
+	 */
+	protected function is_response_loaded_from_cache() : bool {
+
+		return $this->response_loaded_from_cache;
+	}
+
+
+	/**
+	 * Loads the response for the current request from the cache, if available.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return array|null
+	 */
+	protected function load_response_from_cache() {
+
+		return get_transient( $this->get_request_transient_key() );
+	}
+
+
+	/**
+	 * Saves the response to cache.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @param array $response
+	 */
+	protected function save_response_to_cache( array $response ) {
+
+		set_transient( $this->get_request_transient_key(), $response, $this->get_request_cache_lifetime());
+	}
+
+
+	/**
+	 * Gets the response data for broadcasting the request.
+	 *
+	 * Adds a flag to the response data indicating whether the response was loaded from cache.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return array
+	 */
+	protected function get_request_data_for_broadcast() : array {
+
+		$request_data = parent::get_request_data_for_broadcast();
+
+		if ( $this->is_request_cacheable() ) {
+			$request_data['force_refresh'] = $this->get_request()->should_refresh();
+			$request_data['should_cache']  = $this->get_request()->should_cache();
+		}
+
+		return $request_data;
+	}
+
+
+	/**
+	 * Gets the response data for broadcasting the request.
+	 *
+	 * Adds a flag to the response data indicating whether the response was loaded from cache.
+	 *
+	 * @since 5.10.10
+	 *
+	 * @return array
+	 */
+	protected function get_response_data_for_broadcast() : array {
+
+		$response_data = parent::get_response_data_for_broadcast();
+
+		if ( $this->is_request_cacheable() ) {
+			$response_data['from_cache'] = $this->is_response_loaded_from_cache();
```

Comment:
Additionally, for this and the request flags, can we bump them up to the top of the array so it's more visible among the other properties?

Reply:
Sure thing